### ISSUE TITLE
Spellchecking in comments

### ISSUE THREAD

**Issue by jxson** [atom/spell-check#19]
It would be a nice option to have the spellcheck enabled for comments. On some projects the comments get generated into docs etc. and this would help correct mistakes on the fly...

**Comment by mesosteros** [comment]
I was just about to ask for this as well. Comments for HTML and JS should be spell checked.
Would be nice if HTML tag content would have it as well.

**Comment by beck** [comment]
:+1:

**Comment by respectTheCode** [comment]
This would be really helpful.

**Comment by mkantor** [comment]
I tried what I thought was the obvious user-level fix: adding "comment.line.double-slash.js" and some related scopes to the list of grammars in spell-check's settings, but it doesn't work (and there's no obvious error message that I can see, either). Adding "source.js" does work, however, which confuses me.

If this ends up being implemented using the current grammar list setting, I have a feature request: allow regex or glob-like patterns (e.g. "comment.*") in the grammar list so that it doesn't have to be specified for each specific language/scope individually.

**Comment by cptaffe** [comment]
:+1: Just came here to ask for this as well. Googling for correct spelling is bothersome.

**Comment by tombh** [comment]
:+1:

**Comment by caillou** [comment]
:+1: Being unable to write more than two words without a typo, this would make my co-workers happy.

**Comment by int3h** [comment]
:+1: for me as well. Sublime has this feature (I can't remember if it's core or a 3rd-party package), and it keeps me from looking like an idiot in my code.

Looking at this package's code, it only grabs the main editor's grammar name to check against the list of scopes the user has configured in spell-check's config. If they match exactly, it spell checks the entire editor.

In other words, there's currently no easy way to tell it to only spell-check a specific scope within a file; it's either the whole file or nothing.

**Comment by stefanRitter** [comment]
:+1: Sublime also spell checks the text nodes of html and haml files, I guess there needs to be a way to combine syntax highlighting and spell checking!?

**Comment by rthijssen** [comment]
+1

**Comment by simonjosefsson** [comment]
:+1: Pinging devs @kevinsawicki, @zcbenz, @nathansobo
Has this been on the back burner intentionally or just been overlooked? (coming up on a year)

It seems to me like a really valuable feature that should be prioritized, thanks for your work!

**Comment by nathansobo** [comment]
I agree this sounds like a very good idea, but it just hasn't been important enough to direct the core team's attention away from other issues. Are there any relevant pull requests we could review?

**Comment by drom** [comment]
+100

**Comment by dengemann** [comment]
Hum. It seems that it's actually working for me, I do get spellchecking in e.g. Python doc strings using the latest Atom version. Maybe this has been addressed in the meantime?

https://www.dropbox.com/s/i0duunkgvxuq936/Screenshot%202015-04-17%2003.08.15.png?dl=0

**Comment by nixpulvis** [comment]
:+1: As someone who shouldn't have passed elementary level spelling, I'd love to see this feature.

**Comment by azdavis** [comment]
+1'ing this. Spellchecking all comments would be nice by default, but at the very least, let us do something like this.
<img width="239" alt="screen shot 2015-07-14 at 10 45 28 am" src="https://cloud.githubusercontent.com/assets/6496454/8676163/c6c8997e-2a15-11e5-9cd8-3951739c5a18.png">

**Comment by maasha** [comment]
+1

**Comment by 9o** [comment]
:+1:

**Comment by s-light** [comment]
:+1:

**Comment by jivinivan** [comment]
:+1:

**Comment by briandavidson** [comment]
:+1:

**Comment by stretchtiberius** [comment]
:+1:

**Comment by mnquintana** [comment]
Thanks for the feedback everyone! However, all these +1s and :+1:s are just adding noise, so to minimize that I'm going to lock this issue for now – we may unlock it once someone starts working on this enhancement to get further feedback.
